Allow error in create-dmg

davemfish opened the pull request:
There's consistently an error from `create-dmg` when it tries to unmount a volume. Until that's fixed upstream, we can make sure our workflows continue past this step.

Fixes #984 

## Checklist
- [ ] Updated HISTORY.rst (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

emlys commented:
@davemfish this is great. I wonder if we might also add a shorter timeout to the dmg building step? Looks like it took over 30 minutes, I assume most of that is waiting for the detach which is going to fail anyway.

davemfish commented:
> @davemfish this is great. I wonder if we might also add a shorter timeout to the dmg building step? Looks like it took over 30 minutes, I assume most of that is waiting for the detach which is going to fail anyway.

Great idea! Looking back at past builds, I saw some taking ~10 minutes. So I set the timeout for `make binaries` at 15.

emlys commented on .github/workflows/build-and-test.yml:
```diff
@@ -376,6 +376,10 @@ jobs:
+        # Due to a create-dmg issue, sometimes we wait a very long time
+        # for a disk to unmount, and it fails to do so. Instead of waiting
+        # for an inevitable failure, set timeout-minutes.
+        timeout-minutes: 15
```
```suggestion
        timeout-minutes: 15
        continue-on-error: true
```

emlys reviewed (changes requested):
Ah I guess the timeout causes it to fail at the workflow level, maybe `continue-on-error` will fix it?

davemfish commented:
> Ah I guess the timeout causes it to fail at the workflow level, maybe `continue-on-error` will fix it?

Ah thanks for noticing that. I added `continue-on-error` and removed the Makefile changes so we're only addressing this in one place. Presumably we'll always timeout now before `create-dmg` raises the error anyway.

davemfish commented:
Woah 30 mins to restore the node modules cache for Mac, 30 seconds for Windows!
